### PR TITLE
Refactor BEMTREE

### PR DESCRIPTION
**Pull request by tadatuta**
// cc @tavriaforever @gela-d @tormozz48

### CODE REVIEW

**Comment by gela-d** [comment]
:+1: thank u!
There is some style-corrections, but I'll fix it in PR abput codestyle.

And we also need it in `leveldb`.

File: src/blocks/common.blocks/post/_view/post_view_full.bemtree [hunk 1]
```diff
@@ -1,51 +1,52 @@
...
-                lang: this.data.lang
-            }),
+            apply('categories', { ctx: {
+                breadcrumbs: node.breadcrumbs,
+                lang: data.lang
+            } }),
 
-            apply('title', this.ctx.content = source),
-            apply('github', this.ctx = {
+            apply('title', { 'ctx.content' : source }),
```
Comment (tavriaforever) [inline src/blocks/common.blocks/post/_view/post_view_full.bemtree]: white space after colon

File: src/blocks/common.blocks/post/_view/post_view_full.bemtree [hunk 2]
```diff
@@ -1,51 +1,52 @@
...
+            apply('github', { ctx : {
```
Comment (tavriaforever) [inline src/blocks/common.blocks/post/_view/post_view_full.bemtree]: white space after colon

File: src/blocks/common.blocks/post/_view/post_view_full.bemtree [hunk 3]
```diff
@@ -1,51 +1,52 @@
...
                 content: source,
                 node: node
-            }),
-            apply('deps', this.ctx.content = source.deps),
-            apply('contentFull', this.ctx.content = source.content),
-            apply('issue', this.ctx.content = source.repo)
+            } }),
+            apply('deps', { 'ctx.content' : source.deps }),
```
Comment (tavriaforever) [inline src/blocks/common.blocks/post/_view/post_view_full.bemtree]: white space after colon

File: src/blocks/common.blocks/post/_view/post_view_full.bemtree [hunk 4]
```diff
@@ -1,51 +1,52 @@
...
+            apply('contentFull', { 'ctx.content' : source.content }),
```
Comment (tavriaforever) [inline src/blocks/common.blocks/post/_view/post_view_full.bemtree]: white space after colon

File: src/blocks/common.blocks/post/_view/post_view_full.bemtree [hunk 5]
```diff
@@ -1,51 +1,52 @@
...
+            apply('issue', { 'ctx.content' : source.repo })
```
Comment (tavriaforever) [inline src/blocks/common.blocks/post/_view/post_view_full.bemtree]: white space after colon

**Comment by tavriaforever** [comment]
@tadatuta Are you test it?

**Comment by tadatuta** [comment]
@tavriaforever worked well for `bem-site-engine`. I tried not to change logic just rewrite old syntax and improve some small parts. Still I haven't checked it on bem.info / Lego

File: src/blocks/common.blocks/post/_view/post_view_full.bemtree [hunk 6]
```diff
@@ -1,51 +1,52 @@
...
+        return {
+            elem: 'author',
+            authors: this.ctx.authors,
+            translators: this.ctx.translators
+        };
+    }),
 
     //date of creation and last modification
-    date: {
+    mode('data')(function () {
```
Comment (gela-d) [inline src/blocks/common.blocks/post/_view/post_view_full.bemtree]: `data` -> `date`

**Comment by gela-d** [comment]
This tree on bem-info is ok.

**Comment by gela-d** [comment]
I can't check our site with this commit, because of impossibility of cherry-pick this commit to leveldb

**Comment by tadatuta** [comment]
up

**Comment by tadatuta** [comment]
@gela-d https://github.com/bem/bem-site-engine/pull/86

**Comment by gela-d** [comment]
Lets rebase & merge

**Pull request merged**